src/browser/modules/D3Visualization/mapper.ts

```diff
@@ -20,22 +20,22 @@
 
 import { optionalToString } from 'services/utils'
 import Graph from './lib/visualization/components/graph'
-import Node from './lib/visualization/components/node'
+import Node from './lib/visualization/components/Node'
 import Relationship from './lib/visualization/components/relationship'
 
 const mapProperties = (_: any) => Object.assign({}, ...stringifyValues(_))
 const stringifyValues = (obj: any) =>
   Object.keys(obj).map(k => ({ [k]: optionalToString(obj[k]) }))
 
-export function createGraph(nodes: any, relationships: any) {
+export function createGraph(nodes: Node[], relationships: any) {
   const graph = new Graph()
   graph.addNodes(mapNodes(nodes))
   graph.addRelationships(mapRelationships(relationships, graph))
   ;(graph as any).display = { initialNodeDisplay: 300, nodeCount: 1 }
   return graph
 }
 
-export function mapNodes(nodes: any) {
+export function mapNodes(nodes: any): Node[] {
   return nodes.map(
```

Comment:
I'm not sure I understand what this mapping function does. Since it's returning Node[] but is seems to also be taking Node[], when called by createGraph. Is there a type error hidden here somewhere?

Reply:
@OskarDamkjaer - this is the correct type for `createGraph`. In this method, it calls `graph.addNodes(mapNodes(nodes))` assigns mapped `Node[]` data to it's class variable `_nodes`.

https://github.com/neo4j/neo4j-browser/blob/947678af576733051414d286cb16e3fa75ba062b/src/browser/modules/D3Visualization/mapper.ts#L30-L36

https://github.com/neo4j/neo4j-browser/blob/947678af576733051414d286cb16e3fa75ba062b/src/browser/modules/D3Visualization/lib/visualization/components/graph.ts#L74-L82

Reply:
`nodes` in `graph.addNodes(mapNodes(nodes))` is declared to be of type `Node[]` on line 30, so that we should be able to remove the  "any" from `addNodes`? When I do so I get type errors
<img width="487" alt="CleanShot 2021-12-21 at 15 50 30@2x" src="https://user-images.githubusercontent.com/10564538/146949753-a5aca31e-d7e7-4876-b44c-acbf9bab098e.png">

Reply:
So to me it seems there are multiple shapes a node can have, and the point of "mapNodes" is to map from one representation of nodes to another.

Reply:
Looking into it, it seems that the mapping function is meant to take the type of node returned from `extractNodesAndRelationshipsFromRecordsForOldVis` and map it into our own Node class

Reply:
Digging deeper, the nodes passed to the createGraph constructor come from the state in VisualizationView.tsx

Reply:
So they aren't of type "Node", I'm thinking the best way forward is either to make sure everything we say has the `Node` type really is a Node of leave it as any

Reply:
@OskarDamkjaer - it should be like
```ts
// graph.ts
  addNodes(nodes: Node[]) { 
    // ...
 } 
```
But for `mapNodes()`, the parameter has different shape and it's mapped to `Node[]` as the returned value:
```ts
// mapper.ts
  mapNode(nodes: UnmappedNodeFromQueryResult[]): Node[] {
    // ...
  }
```

Reply:
@OskarDamkjaer - thanks for pointing it out. I've distinguished two different `Node` types in the latest commit!